stake-pool/single-pool/src/processor.rs

```diff
@@ -0,0 +1,1079 @@
+            &[pool_stake_info.clone(), system_program_info.clone()],
+            stake_signers,
+        )?;
+
+        invoke_signed(
+            &system_instruction::assign(pool_stake_info.key, stake_program_info.key),
+            &[pool_stake_info.clone(), system_program_info.clone()],
+            stake_signers,
+        )?;
+
+        invoke_signed(
+            &stake::instruction::initialize_checked(pool_stake_info.key, &authorized),
+            &[
+                pool_stake_info.clone(),
+                rent_info.clone(),
+                pool_authority_info.clone(),
+                pool_authority_info.clone(),
+            ],
+            authority_signers,
+        )?;
+
+        // delegate stake so it activates
+        invoke_signed(
+            &stake::instruction::delegate_stake(
+                pool_stake_info.key,
+                pool_authority_info.key,
+                validator_vote_info.key,
+            ),
+            &[
+                pool_stake_info.clone(),
+                validator_vote_info.clone(),
+                clock_info.clone(),
+                stake_history_info.clone(),
+                stake_config_info.clone(),
+                pool_authority_info.clone(),
+            ],
+            authority_signers,
+        )?;
+
+        Ok(())
+    }
+
+    #[inline(never)] // needed to avoid stack size violation
+    fn process_deposit_stake(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        vote_account_address: &Pubkey,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let pool_stake_info = next_account_info(account_info_iter)?;
+        let pool_authority_info = next_account_info(account_info_iter)?;
+        let pool_mint_info = next_account_info(account_info_iter)?;
+        let user_stake_info = next_account_info(account_info_iter)?;
+        let user_token_account_info = next_account_info(account_info_iter)?;
+        let user_lamport_account_info = next_account_info(account_info_iter)?;
+        let clock_info = next_account_info(account_info_iter)?;
+        let clock = &Clock::from_account_info(clock_info)?;
+        let stake_history_info = next_account_info(account_info_iter)?;
+        let token_program_info = next_account_info(account_info_iter)?;
+        let stake_program_info = next_account_info(account_info_iter)?;
+
+        check_pool_stake_address(program_id, vote_account_address, pool_stake_info.key)?;
+        let bump_seed = check_pool_authority_address(
+            program_id,
+            vote_account_address,
+            pool_authority_info.key,
+        )?;
+        check_pool_mint_address(program_id, vote_account_address, pool_mint_info.key)?;
+        check_token_program(token_program_info.key)?;
+        check_stake_program(stake_program_info.key)?;
+
+        // we enforce that our stake is active so depositing/withdrawing during the warmup epoch is impossible
```

Comment:
I'm wondering if we need to enforce that... are there any issues with moving stake during the warmup epoch?

Reply:
yea im gonna think on this again but i believe youre right. im pretty sure i added the "active" qualifier to the function because i didnt _need_ the extra functionality, but i dont believe it does anything bad to have it